### ISSUE TITLE
For Dual Reporting, we are adding Extra (Illegal) disaggregation columns added when Exporting Data (XLS)

### ISSUE THREAD

**Issue by sethgold8** [unicef/etools-partner-reporting-portal#735]
When Dual Reporting is occurring and we Export (XLS) QPR or HR, we add extra (illegal) dissaggregation columns. 

You can find Dual Reporting in PD reference_number_0 ( https://prp.tivixlabs.com/app/FM/ip-reporting/pd/3/view/reports?page_size=10&page=1 ) as AO in Micronesia 2017.

QPR 2 and HR 2 have the data filled out so you can use those.

Examples shown below

- [ ] For HR if you export XLS, we have too many columns. app/FM/ip-reporting/pd/3/report/80/edit?page=1

Columns have dissagregations subtotals duplicated. For example there are gender: gender subtotals; heigh: height subtotals, gender: gender: height subtotals. Legitimate Column names are duplicated too.

Some examples
![Screen Shot 2018-06-19 at 2.48.04 PM.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59f1189df815e200161f654a/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77256455fa6b7191a37af116234f5f5b5ecc3c4a4356bcbbd671f3e57862804e0ee5823b5777933d12aab1d11eba01ba29894a82b0bb884ad2f42f67b6d5db1063474f4dcbed5727a19e41c95df442e54950a6173062ee702431.png)
![Screen Shot 2018-06-19 at 2.49.58 PM.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59f1189df815e200161f654a/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77256455fa6b7191a37af116224f5a575ecc3c4a4356bcbbd671f3e57862804e0ee5823b5777933d12aab1d11eba01ba29894a82b0bb884ad2f42f67b6d5db1063474f4dcbed5727a19041c95df442e54950a6173062ee742431.png)
![Screen Shot 2018-06-19 at 2.52.32 PM.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59f1189df815e200161f654a/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77256455fa6b7191a37af117294f5c5d5ecc3c4a4356bcbbd671f3e57862804e0ee5823b5777933d12aab1d11eba01ba29894a82b0bb884ad2f42f67b6d5db1063474f4dcbed5727a19041c95df442e54950a6173062e17e2431.png)

Column name is duplicated. In HR 2 as AO (Micronesia 2017 ) app/FM/ip-reporting/pd/3/report/80/edit?page=1

![Screen Shot 2018-06-20 at 4.55.53 PM.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59f1189df815e200161f654a/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77256456f36b7191a37cf1172e4f5a5c5ecc3c4a4356bcbbd671f3e57862804e0ee5823b5777933d12aab1d11eba01ba29894a82b0bb884ad2f42f67b6d5db1063474f4dcbed5727a19e41c95df442e44e5dad1a3362e37e2431.png)

- [ ] Same issue In QPR if you export XLS, we have too many columns. app/FM/ip-reporting/pd/3/report/78/edit?page=1

- [ ] Same issue on Progress reports XLS export app/FM/ip-reporting/progress-reports?status=Due%2COve%2CSen&pd_ref_title=_0&page=1 for AO as Micronesia 2017

- [ ] Same issue on Indicators XLS export app/FM/ip-reporting/indicators?pd_statuses=Act&page_size=10&page=2&pds=3 for AO as Micronesia 2017

**Comment by dfryta** [comment]
I've updated main excel header function. Please check @sethgold8 .

**Comment by sethgold8** [comment]
@dfryta I don't see any with columns with 3 disaggregations so that is good. Can we make so no headers are duplicated though? 

![Screen Shot 2018-06-26 at 5.47.01 PM.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59f1189df815e200161f654a/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77256456f56b7191a37df1162c4f5f5e5ecc3c4a4356bcbbd671f3e57862804e0ee5823b5777933d12aab1d11eba01ba29894a82b0bb884ad2f42f67b6d5db1063474f4dcbed5727a19e41c95df54be14b55a1113762e7762431.png)

**Comment by sethgold8** [comment]
@sethgold8 look for duplicated data both with columsn

**Comment by dfryta** [comment]
@sethgold8 @n1207n  I've dig into this and our report generator is OK. Problem is in fake data - it has different disaggregation in same workspace (https://prp.tivixlabs.com/api/admin/indicator/disaggregation/). As you can see there is non-duplicated data for each of Workspace, but we also have disaggrgations with empty workspace that Micronesia is using:
![Select_disaggregation_to_change___PRP.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/599ea4b9f8cd181455aa3f82/125516c66e82c728ace21e0d46db9196268197968be0ac42a35ae2912e47337a260a9c3f7fbae020be4c7c04303021cc23384d58eeb9cb74fce36f34985619e8d93e097cc56811afbe8b4cbe04ba20824d80b1ba8d4ec6fa2137a5dad11b61435c52dcbd4f3fb2984cc259f348e14f52a01a356ae072316e0a.png)

In other words, duplicated columns are not really duplicated - they have diffrent ID's but with same name. So problem is in fake data itself.

**Comment by sethgold8** [comment]
confirmed
